File: backend/src/main/java/com/scholarzim/util/RoleNames.java
package com.scholarzim.util;


public final class RoleNames {

    public static final String APPLICANT = "ROLE_APPLICANT";
    public static final String PROVIDER = "ROLE_PROVIDER";
    public static final String ADMIN = "ROLE_ADMIN";

    private RoleNames() {
    }

    public static String displayLabel(String roleName) {
        if (roleName == null) {
            return "User";
        }
        return switch (roleName) {
            case APPLICANT -> "Student";
            case PROVIDER -> "Provider";
            case ADMIN -> "Admin";
            default -> roleName.replace("ROLE_", "").toLowerCase();
        };
    }
}

File: app/Support/RoleNames.php
<?php

namespace App\Support;

final class RoleNames
{
    public const APPLICANT = 'ROLE_APPLICANT';
    public const PROVIDER = 'ROLE_PROVIDER';
    public const ADMIN = 'ROLE_ADMIN';

    public const ALL = [self::APPLICANT, self::PROVIDER, self::ADMIN];

    private function __construct()
    {
    }

    public static function displayLabel(?string $roleName): string
    {
        return match ($roleName) {
            self::APPLICANT => 'Student',
            self::PROVIDER => 'Provider',
            self::ADMIN => 'Admin',
            null => 'User',
            default => strtolower(str_replace('ROLE_', '', $roleName)),
        };
    }

    /** Where a freshly authenticated user lands. */
    public static function dashboardUrl(?string $roleName): string
    {
        return match ($roleName) {
            self::ADMIN => '/admin/dashboard',
            self::PROVIDER => '/provider/dashboard',
            self::APPLICANT => '/applicant/dashboard',
            default => '/',
        };
    }
}
